### PR TITLE
Deduce a mask's type from the API response

### PR DESCRIPTION
**Pull request by Vinnl**
The /relayaddresses/ and /domainaddresses/ [now include](https://github.com/mozilla/fx-private-relay/pull/1968) a
`mask_type` field, meaning we no longer need our
`markAsRandomAlias` and `markAsCustomAlias` functions that add a
`type` property.

Note that I've [updated the add-on to inject `mask_type`](https://github.com/mozilla/fx-private-relay-add-on/pull/344) as well,
but I left in some compatibility code to ensure that the order in
which we release the website and add-on does not matter.

How to test: basically nothing should have changed in behaviour compared to `main`, esp w.r.t. local label storage, with both the current version of the add-on and with the updates in https://github.com/mozilla/fx-private-relay-add-on/pull/344. Essentially, it should reduce the surface area for bugs in which the website does not agree with the back-end what the type of a mask is.

- [x] l10n changes have been submitted to the l10n repository, if any.
- [x] I've added a unit test to test for potential regressions of this bug.
- [x] I've added or updated relevant docs in the docs/ directory.
- [x] All UI revisions follow the [coding standards](https://github.com/mozilla/fx-private-relay/blob/main/docs/coding-standards.md), and use Protocol tokens where applicable (see `/static/scss/libs/protocol/css/includes/tokens/dist/index.scss`).
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

File: frontend/src/hooks/localLabels.ts [hunk 1]
```diff
@@ -3,7 +3,9 @@ import { useAddonData } from "./addon";
 import { AliasData, isRandomAlias } from "./api/aliases";
 
 export type LocalLabel = {
+  /** `type` is to allow for a transition period while people migrate to a new add-on version that uses `mask_type` */
```
Comment (Vinnl) [inline frontend/src/hooks/localLabels.ts:6]: i.e. a version that includes this: https://github.com/mozilla/fx-private-relay-add-on/pull/344

File: frontend/src/hooks/localLabels.ts [hunk 2]
```diff
@@ -34,24 +36,30 @@ export function useLocalLabels(): LocalLabelHook | NotEnabled {
...
+    const maskType = isRandomAlias(alias) ? "random" : "custom";
 
     // Replace existing labels for this alias:
     const oldLocalLabel = localLabels.find(
-      (localLabel) => localLabel.id === alias.id && localLabel.type === type
+      (localLabel) =>
+        localLabel.id === alias.id &&
+        (localLabel.mask_type === maskType ||
+          (typeof localLabel.mask_type === "undefined" &&
+            localLabel.type === maskType))
```
Comment (Vinnl) [inline frontend/src/hooks/localLabels.ts:47]: Fallback for add-on versions without https://github.com/mozilla/fx-private-relay-add-on/pull/344 - can be removed when that is rolled out broadly.

**Review by maxxcrawford — APPROVED** [review]
Nice clean-up!

**Comment by Vinnl** [comment]
> I don't feel like I have enough context & knowledge to know the best order of merges + deploys.

I feel that ^.^ The order of these two shouldn't matter, and I've set a reminder for myself to submit to cleanup PRs to remove the backwards-compatibility parts from these two PRs once both have rolled out widely.

**Pull request merged**